python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -271,3 +271,101 @@ def categorical_embedding_with_vocabulary_file(feature_tensor, feature_info, fil
     )
 
     return embedding
+
+
+def categorical_indicator_with_vocabulary_file(feature_tensor, feature_info, file_io: FileIO):
+    """
+    Converts a string tensor into a categorical one-hot representation.
+    Works by using a vocabulary file to convert the string tensor into categorical indices
+    and then converting the categories into one-hot representation.
+
+    Args:
+        feature_tensor: String feature tensor
+        feature_info: Dictionary representing the configuration parameters for the specific feature from the FeatureConfig
+
+    Returns:
+        Categorical one-hot representation of input feature_tensor
+
+    Args under feature_layer_info:
+        vocabulary_file: str; path to vocabulary CSV file for the input tensor
+        num_oov_buckets: int; number of out of vocabulary buckets/slots to be used to
+                         encode strings into categorical indices
+
+    NOTE:
+    The vocabulary CSV file must contain two columns - key, id,
```

Comment:
Could you give an example? Why do we need an id? Couldn't the vocabulary file just be:
key, othercolumns_we_ignore <== header
"AAA", ...
"BBB", ...
"CCC", ...
"BBB", ...
"EEE", ...
...

?

Reply:
You don't. If there is no id, then it autogenerates. Check the next line in the comment.

We can use Id to map multiple categories into the same ID, say from a preprocessing job you've found that these categories are similar, etc.